Source/SwiftLintFramework/Rules/Idiomatic/NoGroupingExtensionRule.swift

```diff
@@ -28,12 +29,37 @@ public struct NoGroupingExtensionRule: OptInRule, ConfigurationProviderRule, Aut
 
         let susceptibleNames = Set(elements.compactMap { $0.kind != .extension ? $0.name : nil })
 
-        return elements
-            .filter { $0.kind == .extension && susceptibleNames.contains($0.name) }
-            .map {
-                StyleViolation(ruleDescription: type(of: self).description,
-                               severity: configuration.severity,
-                               location: Location(file: file, byteOffset: $0.offset))
+        return elements.compactMap { element in
+            guard element.kind == .extension, susceptibleNames.contains(element.name) else {
+                return nil
             }
+
+            guard !hasWhereClause(element: element, file: file) else {
+                return nil
+            }
+
+            return StyleViolation(ruleDescription: type(of: self).description,
+                                  severity: configuration.severity,
+                                  location: Location(file: file, byteOffset: element.offset))
+        }
+    }
+
+    private func hasWhereClause(element: NamespaceCollector.Element, file: File) -> Bool {
```

Comment:
nitpick: I'd pass `[String: SourceKitRepresentable]` instead of `NamespaceCollector.Element` to avoid all the `.dictionary` calls

Reply:
Done